dashboard/app/models/plc/enrollment_unit_assignment.rb

```diff
@@ -40,6 +40,8 @@ def enroll_user_in_unit_with_learning_modules(learning_modules)
       learning_modules |=  self.plc_course_unit.plc_learning_modules.where(required: true)
 
       learning_modules.each do |learning_module|
+        next unless learning_module.plc_course_unit == self.plc_course_unit
```

Comment:
I don't want someone to post to the controller with an arbitrary list of learning module ids that might be from various sources. This is to ensure that all the learning modules we are enrolling you in are in this course unit.

I'll add a comment